### PR TITLE
Restrict DocDB names to 24 chars

### PR DESCRIPTION
**Pull request by vishwam**
DocDB’s documentation says it allows 3-50 chars, but their actual length limit varies per-region (bug 846765). Restrict to 24 chars based on their dev’s recommendation.

### CODE REVIEW

**Comment by azurecla** [comment]
Hi __@vishwam__, I'm your friendly neighborhood Azure Pull Request Bot (You can call me AZPRBOT). Thanks for your contribution!
    <p>
        It looks like you're working at Microsoft (vishwams). If you're full-time, we DON'T require a contribution license agreement.
    </p>
    <p>
        If you are a vendor, DO please sign the electronic contribution license agreement. It will take 2 minutes and there's no faxing! https://cla.azure.com.
    </p>

TTYL, AZPRBOT;

**Comment by joyhui** [comment]
@vishwam 
Can you put a comment to the line you fixed? So that we know the reason why we have this 19 in code.
BTW, you mentioned 24 chars is recommended by their dev, why 19 comes out in the coe?

File: Common/Deployment/DeploymentLib.ps1 [hunk 1]
```diff
@@ -234,6 +234,7 @@ function ValidateResourceName()
         "microsoft.documentdb/databaseaccounts"
         {
             $resourceUrl = $global:docdbSuffix
+            $resourceBaseName = $resourceBaseName.Substring(0, [System.Math]::Min(19, $resourceBaseName.Length))
```
Comment (joyhui) [inline Common/Deployment/DeploymentLib.ps1]: Please put a comment here to explain why we have this fix, and how 19 comes out.

Comment (vishwam) [inline Common/Deployment/DeploymentLib.ps1]: Added comment.

**Comment by vishwam** [comment]
@joyhui I've added the comment

**Comment by joyhui** [comment]
Thanks!

**Pull request merged**